src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -688,6 +694,35 @@ public string AccessToken
             }
         }
 
+        /// <summary>
+        ///
+        /// </summary>
+        public Func<AzureADTokenRequestContext, CancellationToken, Task<SqlAuthenticationToken>> AccessTokenCallback
```

Comment:
Please also include a sample on how to use this API in the [samples](https://github.com/dotnet/SqlClient/tree/main/doc/samples) folder demonstrating basic working example, which can be later added to public documentations.

/cc @DavoudEshtehari @David-Engel